packages/liveblocks-core/src/__tests__/_utils.ts

```diff
@@ -123,6 +123,21 @@ export class MockWebSocketServer {
     return this.current.server;
   }
 
+  /**
+   * In 99.9% of cases, the server messages are going to be JSON-serialized
+   * values. If you need to look at the raw message (the 0.1% case), you can
+   * use this.receivedMessagesRaw for that.
+   */
+  public get receivedMessages(): Json[] {
+    return this.receivedMessagesRaw.map((raw) => {
+      try {
+        return JSON.parse(raw) as Json;
+      } catch {
+        return "<non-JSON value>";
```

Comment:
My only worry with it would be that `raw = 'ping'` would not be valid JSON, but `raw = '"ping"'` would, and I thought that could be confusing if that'd happen implicitly. I used this explicit string value to recognize this edge case, rather than hide it. That said, I haven't seen this code path taken just yet in all the tests I've rewritten so far, so perhaps we could even throw in this case.